### PR TITLE
#[133092183] Tasker should be able to accept or reject a quote

### PR DESCRIPTION
**Pull request by victor-kironde**
#### What does this PR do?
Create a table ```quotes```
Create a model for the table ```quotes```
Create a quote notification.
Create a view to capture Tasker response to a Quote
Create a view to show artisan information if the quote is accepted.

#### What are the relevant pivotal tracker stories?
133092183

### CODE REVIEW

File: app/assets/javascripts/quote_notification.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  };
+};
+
+function quotation(action,id){
+    $.ajax({
+      url: ("/dashboard/notifications/" + id),
+      method: "PUT",
+      data: responseData(action)
+    }).then(function(data){
+      // console.log(data)
```
Comment (andela-fadebayo) [inline app/assets/javascripts/quote_notification.js]: Remove `console.log`

File: app/assets/javascripts/quote_notification.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      $('#quote_accept').hide()
+      $('#quote_reject').hide()
+      $('#information').show()
+    },
+    function(req, status, error) {
+      console.log(error)
```
Comment (andela-fadebayo) [inline app/assets/javascripts/quote_notification.js]: Handle this error, probably display it to the user. Remove `console.log`

File: app/controllers/notifications_controller.rb [hunk 3]
```diff
@@ -49,5 +52,6 @@ def set_notification_attr(params)
     @notifiable_attr_to_update = params[:notifiable_attr_to_update]
     @message = params[:message]
     @reply_to_sender = params[:reply_to_sender]
+    @event_name = params[:event_name]
```
Comment (andela-fadebayo) [inline app/controllers/notifications_controller.rb:55]: why is the params retrieved this way?

File: app/models/notification.rb [hunk 4]
```diff
@@ -20,6 +20,10 @@ def self.update_as_notified(user)
       update_all(user_notified: true)
   end
 
+  def self.quote_notification(user)
```
Comment (andela-fadebayo) [inline app/models/notification.rb]: What does this method do?

File: app/models/service.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
+class Service < ActiveRecord::Base
```
Comment (andela-fadebayo) [inline app/models/service.rb:1]: This looks like the same class in @andela-amagana PR. Did you merge his work here and will this result in conflict later?

File: app/views/notifications/index.html.erb [hunk 6]
```diff
@@ -19,25 +19,42 @@
 
       <% elsif @notifications.present? && current_user.user_type == "tasker" %>
         <div class="notification-feed col s8">
+
           <% @notifications.each do |object| %>
             <div class="viewed feed">
               <img src="<%= object.sender.image_url %>" alt="" />
               <p class="title"> <%= object.message %> </p>
-              <a href="#modal1" onclick="modal_object(<%= object.sender.to_json %>, <%= object.id %>)" class="btn info_btn modal-trigger"> view information</a>
+              <%if object.notifiable_type == 'Quote'%>
```
Comment (andela-fadebayo) [inline app/views/notifications/index.html.erb:25]: It's preferred to compare string in one case (either upcase or downcase). It's usually downcase. Compare `object.notifiable_type` against `quote` instead.

File: app/views/notifications/index.html.erb [hunk 7]
```diff
@@ -19,25 +19,42 @@
...
+              <a href="#quote_modal" onclick="quote_object(<%= object.sender.to_json %>,<%=object.notifiable.to_json%>, <%= object.id %>)" class="btn info_btn modal-trigger"> view quote</a>
+
+              <%else%>
+
+              <%#= render partial: "show_information" %>
```
Comment (andela-fadebayo) [inline app/views/notifications/index.html.erb]: What is this else block meant to do? 
Remove comment.

File: app/views/notifications/index.html.erb [hunk 8]
```diff
@@ -19,25 +19,42 @@
...
+              <%end%>
 
             </div>
-          <% end %>
-          <%#= render partial: "show_information" %>
```
Comment (andela-fadebayo) [inline app/views/notifications/index.html.erb:30]: Remove comment if not used.

File: app/views/notifications/index.html.erb [hunk 9]
```diff
@@ -19,25 +19,42 @@
...
+                  <h3 class="">Quote: <span id="quote"></span></h3>
+
+                  <div id= "information">
+                    <h5>Details</h5>
+                    <p class="artisan_phone">Phone: <span id="phone"></span></p>
+                    <p class="artisan_email">Email: <span id="email"></span></p>
+                    <a href="#!" class=" modal-action modal-close waves-effect waves-green btn-flat">Close</a>
+                  </div>
+
+                  <button id='quote_accept' class="btn teal modal-action" onclick="quotation('accepted',<%=object.notifiable.id%>)">Accept</button>
```
Comment (andela-fadebayo) [inline app/views/notifications/index.html.erb]: There should be appropriate spacing, like so
`onclick="quotation('accepted', <%= object.notifiable.id %>)"`
here and other places.

File: spec/factories/services.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+FactoryGirl.define do
+  factory :service do
+    tasker_id 1
+    artisan_id 1
+    skillset_id 1
+    title "MyString"
```
Comment (andela-fadebayo) [inline spec/factories/services.rb]: `MyString` is the title of this service?

File: spec/features/share_contact_spec.rb [hunk 11]
```diff
@@ -36,18 +36,19 @@
...
+  #     Capybara.reset_sessions!
+  #     log_in_with(tasker.email, tasker.password)
+  #     visit notifications_path
+  #     click_on "view quote"
+  #     click_button "Accept"
+  #
+  #     expect(page).to have_content(artisan.firstname)
+  #     expect(page).to have_content(artisan.email)
+  #   end
+  # end
```
Comment (andela-fadebayo) [inline spec/features/share_contact_spec.rb:52]: Why is this commented out?

File: app/models/service.rb [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+            presence: true
+
+  validates :status,
+            presence: true
+
+  validate :end_date_must_be_greater_than_start_date
+
+  enum status: [:unassigned, :assigned, :accepted]
+
+  def self.pending_requests(current_user)
```
Comment (andela-fadebayo) [inline app/models/service.rb:34]: This is not tested

**Review by andela-fadebayo — CHANGES_REQUESTED** [review]

File: app/views/notifications/index.html.erb (hunk 6 above)
Comment (victor-kironde) [reply app/views/notifications/index.html.erb:25]: `Quote` here refers to a class name and should therefore be uppercase.

File: app/views/notifications/index.html.erb (hunk 8 above)
Comment (victor-kironde) [reply app/views/notifications/index.html.erb:30]: Done

File: app/views/notifications/index.html.erb (hunk 9 above)
Comment (victor-kironde) [reply app/views/notifications/index.html.erb]: Done

File: spec/features/share_contact_spec.rb (hunk 11 above)
Comment (victor-kironde) [reply spec/features/share_contact_spec.rb:52]: It was a feature test expecting a tasker to view an artisan's contact information before accepting or rejecting a quote.

**Review by andela-fadebayo — APPROVED** [review]

**Pull request merged**